snuba/settings/__init__.py

```diff
@@ -31,16 +31,37 @@
 # Clickhouse Options
 CLICKHOUSE_MAX_POOL_SIZE = 25
 
-# The number of physical partitions that we will need to map resources to
-# for storage partitioning
-LOCAL_SLICES = 1
-# Mapping of logical (key) to physical (value) partitions for storages
-# that are partitioned in Snuba resources
-LOGICAL_PARTITION_MAPPING: Mapping[str, int] = {
-    str(x): 0 for x in range(0, SENTRY_LOGICAL_PARTITIONS)
+# Mapping storage set key to a mapping of logical partition
+# to slice id
+LOGICAL_PARTITION_MAPPING: Mapping[str, Mapping[int, int]] = {
+    storage_set: {x: 0 for x in range(0, SENTRY_LOGICAL_PARTITIONS)}
+    for storage_set in {
+        "cdc",
+        "discover",
+        "events",
+        "events_ro",
+        "metrics",
+        "migrations",
+        "outcomes",
+        "querylog",
+        "sessions",
+        "transactions",
+        "transactions_ro",
+        "transactions_v2",
+        "errors_v2",
+        "errors_v2_ro",
+        "profiles",
+        "functions",
+        "replays",
+        "generic_metrics_sets",
+        "generic_metrics_distributions",
+    }
```

Comment:
Is there a reason why we're doing this mapping by default, rather than only specifying it for sliced storages?

Reply:
Right, this was something I wanted to clarify as well. Before, it was a general mapping of logical partitions to slices (and this would apply by default to all storage sets). My understanding was that same default meaning needs to be preserved with this new mapping, so this is why I included all of the storage sets. Let me know if I misunderstood something, though.

Reply:
I think we can preserve this functionality without building the mapping explicitly. e.g. not all storage sets will be present in `LOGICAL_PARTITION_MAPPING`, code that relies on `LOGICAL_PARTITION_MAPPING` can assume any non-present keys only have a single slice

Reply:
Ok, that makes sense. I can make it such that `LOGICAL_PARTITION_MAPPING` is empty in default settings.